pkg/cloud-provider/cloudapi/azure/azure_resourcegraph_vmTable.go

```diff
@@ -104,8 +106,48 @@ const (
 		"| project id, name, properties, status=properties.extended.instanceView.powerState.code, networkInterfaces, tags, vnetId"
 )
 
+func ToTimeHookFunc() mapstructure.DecodeHookFunc {
+	return func(
+		f reflect.Type,
+		t reflect.Type,
+		data interface{}) (interface{}, error) {
+		if t != reflect.TypeOf(time.Time{}) {
+			return data, nil
+		}
+
+		switch f.Kind() {
+		case reflect.String:
+			return time.Parse(time.RFC3339, data.(string))
+		case reflect.Float64:
+			return time.Unix(0, int64(data.(float64))*int64(time.Millisecond)), nil
+		case reflect.Int64:
+			return time.Unix(0, data.(int64)*int64(time.Millisecond)), nil
+		default:
+			return data, nil
+		}
+	}
+}
+
+// customDecode use a timeHookFunc() to convert *time.time to desired format.
+// Cannot use 'mapstructure.decode' directly, since it does not understand the type *time.time
+// 'Properties.TimeCreated' expected a map, got 'string' 0xc0003ae060
+// Use the fix suggested in the issue: https://github.com/mitchellh/mapstructure/issues/159
+func customDecode(input map[string]interface{}, result interface{}) error {
```

Comment:
Is this from sample example in sdk ?Doesnt SDK provide it

Reply:
I think so, lets confirm with @Anandkumar26 .

Reply:
There are no examples to show how to parse the output.
Here is a sample examples from resource client, which does not show how to parse the response.
https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/resourcegraph/armresourcegraph/ze_generated_example_resourcegraphclient_client_test.go

I have tried to keep the changes minimal, so that existing data structures can be used.